utils/test/i18n/i18n.util.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { ReactNode, ReactElement } from 'react';
+import { isElement } from 'react-dom/test-utils';
+
+const stringify: (node: ReactNode) => string = (node) => {
+  if (node === undefined || node === null) {
+    return '';
+  }
+
+  if ((node as ReactElement).props) {
```

Comment:
I'm interested to know why this needs to be casted?

Reply:
If it definitely is needed, could a casted variable be saved and re-used instead of casting each time

Reply:
From my research, this is the way to do `instanceof` checks in TS - plus I think it would be more memory efficient than creating casted variables?

Reply:
`node` is typed as a `ReactNode` (on 7), then used as a `ReactElement` here. The `as` is a cast, but as a hint to TS. You could do `const foo = {} as ReactElement` and from a TS POV call/use foo as if it were a `ReactElement` (when it is not). This is needed when TS cannot (despite best effort) figure out what X is.

Do you know what the relationship is between `ReactElement` and `ReactNode` @nictownsend ? I assume `ReactElement` is a `ReactNode`, but we props are only typed on the ReactElement?

Reply:
Exactly that. Function components `children` are `ReactNode` - which can be string, number, boolean, ReactNodeArray, ReactElement. So to get props, you have to be working against a ReactElement type.

I'm using the `as` rather than the cast to avoid new object creation but if we think it cleaner I'm happy to do the reassign cast.

Reply:
Tell me if it wont work, but why not treat `node` as a ReactElement from line 7 (ie change the typing of the function)?